attachment_preview/static/src/js/web_views/form_renderer.esm.js

```diff
@@ -0,0 +1,43 @@
+/** @odoo-module **/
+import {onMounted, onWillUnmount} from "@odoo/owl";
+import {AttachmentPreviewWidget} from "../attachmentPreviewWidget.esm";
+import {FormRenderer} from "@web/views/form/form_renderer";
+import {bus} from "web.core";
+import {patch} from "@web/core/utils/patch";
+
+patch(FormRenderer.prototype, "attachment_preview.FormRenderer", {
+    attachmentPreviewWidget: null,
+
+    setup() {
+        var res = this._super(...arguments);
+        this.attachmentPreviewWidget = new AttachmentPreviewWidget(this);
```

Comment:
At the moment we have two different functionalities for the PDF preview on forms:

- standard Odoo PDF preview
- this module (`attachment_preview`)

Would be nice to have an option to disable this attachment preview on forms for this module (`attachment_preview`). This is a requirement for one of our customers.  So the idea is that the admin could decide to configure the option (enable/disable). This way, when the option is disabled, the standard Odoo preview will be work in place of this module (`attachment_preview`).

Disabling this option can be easily achieved by returning `res` just after the `_super(..)`.

```javascript
        var res = this._super(...arguments);
        if (is_attachment_preview_disabled) return res;
```

Reply:
Yea makes perfect sense. To take it even further, I did ponder whether this module could entirely re-use components added by Odoo to implement their preview feature...

I won't be able to get to this until at least next week.

Reply:
@houzefa-abba did you get time to get to this or you plan to do it in the near future ? thanks for this PR anyway !

Reply:
Hi this is still in my TODO but it has moved further back 😅
Very unlikely I could get back to this PR until end of May

This per-form setting hinges between fix (imposed by odoo16 adding its own preview system) and evolution; perhaps we could add it as a separate evolution PR so this initial PR can get moving as is?

Reply:
Hi I got back onto this topic for a bit

I just rebased on top of latest 16.0 and added a commit to disable Odoo's default attachment sidebar can show up in views with `<div class="o_attachment_preview" />`

However I kept pondering @astirpe's idea above but I can't see a way it fully makes sense:
* **Global setting**: would be useful to disable our viewer in forms in favor of odoo's own viewer when one still wants to use our viewer somewhere else, but it doesn't make much sense to me: such an odoo app would use 2 different viewers based on view type, potentially displaying the same documents in different ways depending on view type.
* **Per-form setting** (on the `<form>` tag): again, I can't see a use case, especially when we assume our viewer is similar (and in parts more powerful) than odoo's base one.

Could we think about it some more before confirming we need such a setting? In a separate PR possibly?

Thanks